pymatviz/structure_viz/plotly.py

```diff
@@ -4,11 +4,9 @@
 
 import math
 import warnings
-from collections.abc import Callable, Sequence
-from typing import TYPE_CHECKING, Any, Literal
```

Comment:
No problem at all :)

Putting aside the import time, it appears our import linter is not doing its job?
- Some completely unused imports are not removed, like this one.
- Some type checking imports are not relocated.
- Even duplicate imports are not removed (import both at runtime and for type checking)

Reply:
didn't encounter `ruff` import sorting issues yet. is it still happening? can you give a specific example?

Reply:
Yes I'm not seeing `ruff` errors from main branch either (when there should):
```
(venv) yang@Yang-MacBook pymatviz % ruff --version
ruff 0.7.0
(venv) yang@Yang-MacBook pymatviz % ruff check . --select I001
All checks passed!
(venv) yang@Yang-MacBook pymatviz % git branch
  count-elems-default-fill-val
* main
```

- Some completely unused imports are not removed
I must have misread here, they should be duplicate imports

- Some type checking only imports are not relocated
Check the `pymatviz/coordination.py` change in this PR, `ruff` doesn't report an error even with `from __future__ import annotations`:

<img width="1059" alt="image" src="https://github.com/user-attachments/assets/1cca75c3-1c03-47d8-a0fa-1bca70def4b6">

- Even duplicate imports are not removed (import both at runtime and for type checking)

It's everywhere in this PR (check the rest of the diff)